content/docs/digging_deeper/tuyau_client.md

```diff
@@ -0,0 +1,320 @@
+
+Ensure your package manager or monorepo tool can resolve the `workspace:*` syntax. If not, use the appropriate syntax for your tool.
+
+:::
+
+## Initialize the client
+
+In your frontend project, create the Tuyau client by importing the API definition:
+
+```ts
+import { createTuyau } from '@tuyau/client';
+import { api } from '@acme/server/api';
+
+export const tuyau = createTuyau({
+  api,
+  baseUrl: 'http://localhost:3333',
+});
+```
+
+Here, api is a runtime object containing both the API definition (as a type) and the routes of your API. This setup allows you to map route names to paths and ensures type safety when calling your routes.
+
+If you prefer not to include the runtime code for route names, you can import only the `ApiDefinition` type:
+
+```ts
+import { createTuyau } from '@tuyau/client';
+import type { ApiDefinition } from '@acme/server/api';
+
+export const tuyau = createTuyau<{ definition: ApiDefinition }>({
+  baseUrl: 'http://localhost:3333',
+});
+```
+
+This approach omits the runtime code for route names but still provides type safety when calling your routes by their path (e.g., `tuyau.users.$get()`). However, you will lose the ability to use route helpers like `$has`, `$current`, and `$route`.
+
+## RPC Client
+
+This client is built on top of [Ky](https://github.com/sindresorhus/ky), a lightweight HTTP client, and provides a straightforward interface for making requests to your API endpoints.
+
+With the client instance in place, you can make requests to your API endpoints using method chaining:
+
+```ts
+// GET /users
+await tuyau.users.$get();
+
+// POST /users { name: 'John Doe' }
+await tuyau.users.$post({ name: 'John Doe' });
+
+// PUT /users/1 { name: 'John Doe' }
+await tuyau.users({ id: 1 }).$put({ name: 'John Doe' });
+
+// GET /users/1/posts?limit=10&page=1
+await tuyau.users({ id: 1 }).posts.$get({ query: { page: 1, limit: 10 } });
+```
+
+This approach ensures that all requests are type-safe, with parameters, payloads, query parameters, and responses all being validated at compile time.
+
+If you prefer to use route names instead of paths, you can utilize the `$route` method:
+
+```ts
+// Backend
+router.get('/posts/:id/generate-invitation', '...')
+  .as('posts.generateInvitation');
+
+// Client
+await tuyau
+  .$route('posts.generateInvitation', { id: 1 })
+  .$get({ query: { limit: 10, page: 1 } });
+```
+
+### Path parameters
+
+For routes with path parameters, pass an object to the corresponding function:
+
+```ts
+// Backend
+router.get('/users/:id/posts/:postId/comments/:commentId', '...');
+
+// Client
+const result = await tuyau.users({ id: 1 })
+  .posts({ postId: 2 })
+  .comments({ commentId: 3 })
+  .$get();
+```
+
+### File uploads
+
+When a File instance is passed, Tuyau automatically converts it to a ?`FormData` instance and sets the appropriate headers. The payload is serialized using the `object-to-formdata` package, ensuring that the file is correctly formatted for transmission.
```

Comment:
There is a `?` before `FormData` that seems like a typo.

Also, can we link to `object-to-formdata` package for easier navigation?